### PR TITLE
[CFM-1258-fix] Add unflag functionality when promoting content at group.

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-1258

### CODE REVIEW

**Comment by anvmn** [comment]
out

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module [hunk 1]
```diff
@@ -207,14 +212,27 @@ function c4m_features_og_group_dashboard_form_alter(&$form, &$form_state, $form_
...
     }
   }
 }
 
 /**
+ * Submit handler for the flag confirm form.
+ *
+ * Executes unflagging of content flagged by c4m_og_content_highlight flag.
+ */
+function c4m_features_og_group_dashboard_promote_flag_submit(&$form, &$form_state) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module]: The name of the function is a contradicted to its logic, please change it.
also elaborate the docs.

This logic is not trivial.

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module [hunk 2]
```diff
@@ -192,10 +192,15 @@ function c4m_features_og_group_dashboard_form_alter(&$form, &$form_state, $form_
 
       // Get flagged content.
       $flagged_og_content = c4m_features_og_group_dashboard_get_flagged_group_content($group_id, $flag_id);
-      $flagged_og_content = entity_load($flag->entity_type, array_keys($flagged_og_content));
 
-      // Show summary of entities who will be unflagged if you confirm.
       if (!empty($flagged_og_content)) {
+        // Store the entities to be unflaged, as part of form state.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module]: `...entities IDs to...`

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module [hunk 3]
```diff
@@ -192,10 +192,15 @@ function c4m_features_og_group_dashboard_form_alter(&$form, &$form_state, $form_
...
+        $form_state['flagged_og_content'] = array_keys($flagged_og_content);
+        // Add functions that executes unflaging.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module]: `the unflagging of ...`

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module [hunk 4]
```diff
@@ -192,10 +192,15 @@ function c4m_features_og_group_dashboard_form_alter(&$form, &$form_state, $form_
...
+        $form['#submit'][] = 'c4m_features_og_group_dashboard_promote_flag_submit';
+
+        // Show summary of entities who will be unflagged if you confirm.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module]: `..if the user confirm.`

**Comment by ordavidil** [comment]
@anvmn 
did you also tried to directly demote content from the link ?

**Comment by anvmn** [comment]
@ordavidil 
Yes. It's not related to this this development, though.
Here, when promoting content, we check if promoted content already exists, and demote it, if found.

**Comment by ordavidil** [comment]
@anvmn 
the title of the ticket is:
> Add demote icon for SA/GO/GA

so it is related in here,
does it work as it should ?

**Comment by anvmn** [comment]
@ordavidil 
Yes, it works.

**Pull request merged**